test/util.ts

```diff
@@ -0,0 +1,273 @@
+  return Uint8Array.from(
+    (await generateSeed(signer)).seed
+      .toString()
+      .split(',')
+      .map((x: any) => +x),
+  );
+};
+
+function toHexString(byteArray: any) {
+  return Array.prototype.map
+    .call(byteArray, function (byte) {
+      return ('0' + (byte & 0xff).toString(16)).slice(-2);
+    })
+    .join('');
+}
+
+export const getPublicKey = async (seed: any) => {
+  const seedString = new TextDecoder().decode(seed);
+
+  const publicKey = ZkCrypto.getEddsaPublicKey(seedString);
+  const compressedPublicKey = ZkCrypto.getEddsaCompressedPublicKey(seedString);
+
+  const x = `0x${publicKey.slice(0, 64)}`;
+  const y = `0x${publicKey.slice(64)}`;
+  return {
+    publicKey,
+    compressedPublicKey,
+    x,
+    y,
+  };
+};
+
+export const getAccountNameHash = async (accountName: string) => {
+  return ZkCrypto.getAccountNameHash(accountName);
+};
+
+export const getKeccak256 = (name) => {
+  return ethers.utils.keccak256(ethers.utils.toUtf8Bytes(name));
+};
+
+export const transferFunds = async (signer: any, to: string, amount: string) => {
+  const tx = await signer.sendTransaction({
+    from: await signer.getAddress(),
+    to,
+    value: ethers.utils.parseEther(amount),
+  });
+  await tx.wait();
+};
+
+export interface StoredBlockInfo {
+  blockSize: number;
+  blockNumber: number;
+  priorityOperations: number;
+  pendingOnchainOperationsHash: string;
+  timestamp: number;
+  stateRoot: string;
+  commitment: string;
+}
+
+export interface CommitBlockInfo {
+  newStateRoot: string;
+  publicData: string;
+  timestamp: number;
+  publicDataOffsets: number[];
+  blockNumber: number;
+  blockSize: number;
+}
+
+export function hashStoredBlockInfo(block: StoredBlockInfo) {
+  const encode = ethers.utils.defaultAbiCoder.encode(
+    ['uint16', 'uint32', 'uint64', 'bytes32', 'uint256', 'bytes32', 'bytes32'],
+    [
+      block.blockSize,
+      block.blockNumber,
+      block.priorityOperations,
+      block.pendingOnchainOperationsHash,
+      block.timestamp,
+      block.stateRoot,
+      block.commitment,
+    ],
+  );
+  return ethers.utils.keccak256(encode);
+}
+export enum PubDataType {
+  EmptyTx,
+  RegisterZNS,
+  Deposit,
+  DepositNft,
+  Transfer,
+  Withdraw,
+  CreateCollection,
+  MintNft,
+  TransferNft,
+  AtomicMatch,
+  CancelOffer,
+  WithdrawNft,
+  FullExit,
+  FullExitNft,
+}
+
+export function encodePubData(pubDataType: string[], pubData: any[]) {
+  return ethers.utils.solidityPack(pubDataType, pubData);
+}
+
+export function encodePackPubData(pubDataType: string[], pubData: any[]) {
+  let data = ethers.utils.solidityPack(pubDataType, pubData);
+
+  while (data.length < 121 * 2 + 2) {
+    data += '00';
+  }
+
+  return data;
+}
+
+export function padEndBytes121(data: string) {
+  while (data.length < 121 * 2 + 2) {
+    data += '00';
+  }
+
+  return data;
+}
+
+export const PubDataTypeMap = {
+  [PubDataType.RegisterZNS]: ['uint8', 'uint32', 'bytes20', 'bytes32', 'bytes32', 'bytes32'],
```

Comment:
`accountName` should be a `bytes20`?

https://github.com/bnb-chain/zkbnb-contract/blob/develop/contracts/lib/TxTypes.sol#L38